diff --git a/CMakeLists.txt b/CMakeLists.txt
index 78d54a6..471ab00 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -257,6 +257,7 @@ target_include_directories(apsi PUBLIC
 target_include_directories(apsi PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/sender>
     $<INSTALL_INTERFACE:${APSI_INCLUDES_INSTALL_DIR}>)
+
 apsi_set_version(apsi)
 apsi_link_threads(apsi)
 apsi_install_target(apsi APSITargets)
@@ -273,10 +274,9 @@ if(APSI_USE_ZMQ)
     target_link_libraries(apsi PUBLIC libzmq-static cppzmq-static)
 endif()
 
-# Configurations for FourQlib: system, arch, SIMD, and assembler
-target_compile_options(apsi PUBLIC -DHAVE_CONFIG)
-target_compile_options(apsi PUBLIC -DUSE_SECURE_SEED)
-target_compile_options(apsi PUBLIC -DUSE_ENDO=true)
+# Add FourQlib
+target_include_directories(apsi PUBLIC ${EXT_BUILD_DEPS}/FourQlib/include)
+target_link_libraries(apsi PUBLIC FourQ)
 
 # Set system
 if(MSVC)
@@ -285,59 +285,6 @@ elseif (UNIX)
     target_compile_options(apsi PUBLIC -D__LINUX__)
 endif()
 
-# Detect architecture
-include(DetectArch)
-if(APSI_FOURQ_ARM64)
-    # _ARM64_ needs to be set if the ARM64 optimizations are used
-    # (in UNIX) or the generic implementation is used (Windows)
-    target_compile_options(apsi PUBLIC -D_ARM64_)
-endif()
-if(CMAKE_SYSTEM_PROCESSOR STREQUAL x86)
-    target_compile_options(apsi PUBLIC -D_X86_)
-endif()
-
-if(APSI_FOURQ_AMD64)
-    target_compile_options(apsi PUBLIC -D_AMD64_)
-    message(STATUS "FourQlib optimization: arch=AMD64")
-elseif(APSI_FOURQ_ARM64 AND UNIX)
-    message(STATUS "FourQlib optimization: arch=ARM64")
-else()
-    target_compile_options(apsi PUBLIC -D_GENERIC_)
-    message(STATUS "FourQlib optimization: arch=GENERIC")
-endif()
-
-# Detect AVX instructions
-if(APSI_FOURQ_AMD64 AND (APSI_USE_AVX OR APSI_USE_AVX2))
-    include(FindAVX)
-    check_for_avx(apsi)
-    if(HAVE_AVX2_EXTENSIONS AND APSI_USE_AVX2)
-        target_compile_options(apsi PUBLIC -D_AVX2_)
-        set(APSI_USE_AVX OFF CACHE BOOL ${APSI_USE_AVX_OPTION_STR} FORCE)
-        message(STATUS "FourQlib optimization: simd=AVX2")
-    elseif(HAVE_AVX_EXTENSIONS AND APSI_USE_AVX)
-        target_compile_options(apsi PUBLIC -D_AVX_)
-        set(APSI_USE_AVX2 OFF CACHE BOOL ${APSI_USE_AVX2_OPTION_STR} FORCE)
-        message(STATUS "FourQlib optimization: simd=AVX")
-    endif()
-else()
-    set(APSI_USE_AVX OFF CACHE BOOL ${APSI_USE_AVX_OPTION_STR} FORCE)
-    set(APSI_USE_AVX2 OFF CACHE BOOL ${APSI_USE_AVX2_OPTION_STR} FORCE)
-    message(STATUS "FourQlib optimization: simd=OFF")
-endif()
-
-# Use optimized assembly on UNIX
-if(APSI_USE_ASM AND UNIX AND NOT APPLE AND NOT CYGWIN AND NOT MINGW)
-    check_language(ASM)
-    if(CMAKE_ASM_COMPILER)
-        enable_language(ASM)
-        target_compile_options(apsi PUBLIC -D_ASM_)
-        message(STATUS "FourQlib optimization: asm=ON")
-    endif()
-else()
-    set(APSI_USE_ASM OFF CACHE BOOL ${APSI_USE_ASM_OPTION_STR} FORCE)
-    message(STATUS "FourQlib optimization: asm=OFF")
-endif()
-
 # Add source files to library and header files to install
 # Must follow configurations for FourQlib
 set(APSI_SOURCE_FILES "")
diff --git a/common/apsi/CMakeLists.txt b/common/apsi/CMakeLists.txt
index a65bbfe..60e246e 100644
--- a/common/apsi/CMakeLists.txt
+++ b/common/apsi/CMakeLists.txt
@@ -28,7 +28,6 @@ install(
         ${APSI_INCLUDES_INSTALL_DIR}/apsi
 )
 
-add_subdirectory(fourq)
 add_subdirectory(network)
 add_subdirectory(oprf)
 add_subdirectory(util)
diff --git a/common/apsi/network/zmq/zmq_channel.cpp b/common/apsi/network/zmq/zmq_channel.cpp
index bcaa013..93e2b4a 100644
--- a/common/apsi/network/zmq/zmq_channel.cpp
+++ b/common/apsi/network/zmq/zmq_channel.cpp
@@ -8,7 +8,7 @@
 #include <stdexcept>
 
 // APSI
-#include "apsi/fourq/random.h"
+#include "random.h"
 #include "apsi/log.h"
 #include "apsi/network/result_package_generated.h"
 #include "apsi/network/sop_generated.h"
diff --git a/common/apsi/oprf/ecpoint.cpp b/common/apsi/oprf/ecpoint.cpp
index d12313f..5fd9be1 100644
--- a/common/apsi/oprf/ecpoint.cpp
+++ b/common/apsi/oprf/ecpoint.cpp
@@ -10,10 +10,10 @@
 #include "apsi/util/utils.h"
 
 // FourQ
-#include "apsi/fourq/FourQ.h"
-#include "apsi/fourq/FourQ_api.h"
-#include "apsi/fourq/FourQ_internal.h"
-#include "apsi/fourq/random.h"
+#include "FourQ.h"
+#include "FourQ_api.h"
+#include "FourQ_internal.h"
+#include "random.h"
 
 // SEAL
 #include "seal/randomgen.h"
diff --git a/common/apsi/util/label_encryptor.cpp b/common/apsi/util/label_encryptor.cpp
index 3e00b5e..57a5b40 100644
--- a/common/apsi/util/label_encryptor.cpp
+++ b/common/apsi/util/label_encryptor.cpp
@@ -9,7 +9,7 @@
 #include <vector>
 
 // APSI
-#include "apsi/fourq/random.h"
+#include "random.h"
 #include "apsi/util/label_encryptor.h"
 #include "apsi/util/utils.h"
 
